tests/pgshovel/replication/targets/postgres.py

```diff
@@ -0,0 +1,309 @@
+import pytest
+import psycopg2
+import uuid
+from copy import deepcopy
+from itertools import islice
+from kafka import (
+    KafkaClient,
+    SimpleProducer,
+)
+from kazoo.client import KazooClient
+
+from kafka import KafkaClient
+from tests.pgshovel.fixtures import (
+    cluster,
+    create_temporary_database
+)
+from tests.pgshovel.streams.fixtures import (
+    batch_identifier,
+    make_batch_messages,
+    transaction,
+)
+
+from pgshovel.administration import (
+    create_set,
+    get_managed_databases,
+    initialize_cluster,
+)
+from pgshovel.cluster import Cluster
+from pgshovel.interfaces.common_pb2 import (
+    Column,
+    Row,
+    Snapshot,
+    Tick,
+    Timestamp,
+)
+from pgshovel.interfaces.configurations_pb2 import ReplicationSetConfiguration
+from pgshovel.interfaces.streams_pb2 import (
+    BeginOperation,
+    CommitOperation,
+    MutationOperation,
+    RollbackOperation,
+)
+from pgshovel.relay.streams.kafka import KafkaWriter
+from pgshovel.replication.loaders.simple import SimpleLoader
+from pgshovel.replication.streams.kafka import KafkaStream
+from pgshovel.replication.targets.postgresql import PostgreSQLTarget
+from pgshovel.utilities.conversions import to_snapshot
+
+
+class LimitedKafkaStream(KafkaStream):
```

Comment:
This class is necessary because the `KafkaStream.consume()` method (called internally by the target) builds the generator that is consumed, but the default implementation returns an infinite generator and I needed it to be finite for the test.